File: src/main/java/org/distril/beengine/world/generator/impl/FlatGenerator.java
package org.distril.beengine.world.generator.impl;

import org.distril.beengine.material.Material;
import org.distril.beengine.world.chunk.Chunk;
import org.distril.beengine.world.generator.Generator;

import java.util.concurrent.ThreadLocalRandom;

public class FlatGenerator implements Generator {

	@Override
	public void generate(ThreadLocalRandom random, Chunk chunk) {
		for (int x = 0; x < 16; x++) {
			for (int z = 0; z < 16; z++) {
				chunk.setBlock(x, 0 + 50, z, 0, Material.BEDROCK.getBlock());
				chunk.setBlock(x, 1 + 50, z, 0, Material.DIRT.getBlock());
				chunk.setBlock(x, 2 + 50, z, 0, Material.DIRT.getBlock());
				chunk.setBlock(x, 3 + 50, z, 0, Material.GRASS.getBlock());
			}
		}
	}
}

File: src/main/kotlin/org/distril/beengine/world/generator/impl/FlatGenerator.kt
package org.distril.beengine.world.generator.impl

import org.distril.beengine.material.Material
import org.distril.beengine.world.chunk.Chunk
import org.distril.beengine.world.generator.Generator
import java.util.concurrent.ThreadLocalRandom

class FlatGenerator : Generator {

	override fun generate(random: ThreadLocalRandom, chunk: Chunk) {
		for (x in 0..15) for (z in 0..15) {
			chunk.setBlock(x, 0, z, Material.BEDROCK.getBlock())
			chunk.setBlock(x, 1, z, Material.DIRT.getBlock())
			chunk.setBlock(x, 2, z, Material.DIRT.getBlock())
			chunk.setBlock(x, 3, z, Material.GRASS.getBlock())
		}
	}
}
